src/include/sof/audio/component.h

```diff
@@ -307,6 +307,16 @@ struct comp_ops {
 				   const struct comp_ipc_config *ipc_config,
 				   const void *ipc_specific_config);
 
+	int (*chain_dma_create)(struct comp_dev *dev,
+					uint8_t host_dma_id, uint8_t link_dma_id,
+					uint32_t fifo_size, bool scs);
+
+	int (*chain_dma_remove)(struct comp_dev *dev,
+					uint8_t host_dma_id);
+
+	int (*chain_dma_trigger)(struct comp_dev *dev,
+					uint8_t host_dma_id, int cmd);
```

Comment:
can a component have both - "normal" and chain DMA operations? If not, would it be possible to reuse `.create()`, `.free()`, and `.trigger()`? I see that these function have an additional host_dma_id parameter, but maybe it can be obtained from the device?

Reply:
It is possible. Originally I was thinking about that, however, I decided to treat device instance as manager which contain all information about all possible chains. Finally I don't see any problem to change it to one chain per device.

Reply:
interesting. I don't know much about these chains, what's their life-cycle? Aren't they also just components in pipelines, instantiated from the topology? If yes - then it would be logical to also treat them as normal components - one object per instance?

Reply:
Their lifecycle is independent from pipelines and outside of their scope, but from the logical point of view it is reasonable to treat them as separate components and identify each of them from ipc handler level. Maximum possible number of chains is defined as sum of HDA host inputs + HDA host outputs (platform dependent).